### PR TITLE
storage: update client raft unit tests to remove store gossiper

### LINKED ISSUE #8263: storage: cleanup usage of gossiputil.NewStoreGossiper in client_raft_test.go
There are a number of instances of 
```
for _, g := range mtc.gossips {
	gossiputil.NewStoreGossiper(g).GossipStores(storeDescs, t)
}
```

This seems to be short circuiting the standard gossip system and should investigated. If this is actually needed, we should fix the underlying issue.

Split out from #7684.

### PR DESCRIPTION
**Pull request by BramGruneir**
This change alters the replication unit tests to use the normal pathway for
stores gossiping their descriptors.

Fixes #8263

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/8668)

### CODE REVIEW

**Comment by tamird** [comment]
Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[storage/client_raft_test.go, line 741 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPUNE8TBA8nxOJzw9pE:-KPUNE8TBA8nxOJzw9pF:bopf1da) ([raw file](https://github.com/cockroachdb/cockroach/blob/5d09ed157cfbb6f4d8b7d736d335da6f229fc738/storage/client_raft_test.go#L741)):*
> ```Go
> 	}
> 	for _, g := range mtc.gossips {
> 		gossiputil.NewStoreGossiper(g).GossipStores(storeDescs, t)
> ```

locally i count 4 instances of this pattern, but only two are removed. Why are the other two needed?

---

*[storage/client_test.go, line 373 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPUNfBqy0hOrULK-S0k:-KPUNfBqy0hOrULK-S0l:b-d1nrjz) ([raw file](https://github.com/cockroachdb/cockroach/blob/cdb6d126af4bc49f1bf6b5a020c76c92e9a7175b/storage/client_test.go#L373)):*
> ```Go
> 
> // initGossipNetwork gossips all store descriptors and waits until all
> // storePools have received the those descriptors.
> ```

"the those"

---

*[storage/client_test.go, line 383 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPUNvz0dLTB6nPHzvOY:-KPUNvz0dLTB6nPHzvOZ:b4jea3b) ([raw file](https://github.com/cockroachdb/cockroach/blob/cdb6d126af4bc49f1bf6b5a020c76c92e9a7175b/storage/client_test.go#L383)):*
> ```Go
> 			)
> 			if alive != len(m.stores) {
> 				return errors.Errorf("node %d's store pool only has %d alive stores, expected 3",
> ```

replace "3" with `%d` `len(stores)`

---

*[storage/client_test.go, line 389 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPUO1JU97QnPIVkMENp:-KPUO1JU97QnPIVkMENq:b-wv7ees) ([raw file](https://github.com/cockroachdb/cockroach/blob/cdb6d126af4bc49f1bf6b5a020c76c92e9a7175b/storage/client_test.go#L389)):*
> ```Go
> 		return nil
> 	})
> 	log.Info(context.Background(), "gossip network initialized")
> ```

might remove this

---

*[storage/store_pool.go, line 472 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPUHvMknac6AsYIxnq5:-KPUHvMlPR1oWs9_Fubc:b-lmhsru) ([raw file](https://github.com/cockroachdb/cockroach/blob/cdb6d126af4bc49f1bf6b5a020c76c92e9a7175b/storage/store_pool.go#L472)):*
> ```Go
> // Attr->stores, for efficiency. Ensure that entries in this map still
> // have an opportunity to be garbage collected.
> func (sp *StorePool) GetStoreList(required roachpb.Attributes, deterministic bool) (StoreList, int, int) {
> ```

since this is only exported for tests, consider exporting it via helpers_test.go

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668)*

**Comment by BramGruneir** [comment]
Review status: 1 of 4 files reviewed at latest revision, 5 unresolved discussions, some commit checks pending.

---

*[storage/client_raft_test.go, line 741 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPUNE8TBA8nxOJzw9pE:-KPUV6fkhaNVlM2zmNzn:bpydzlf) ([raw file](https://github.com/cockroachdb/cockroach/blob/5d09ed157cfbb6f4d8b7d736d335da6f229fc738/storage/client_raft_test.go#L741)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> locally i count 4 instances of this pattern, but only two are removed. Why are the other two needed?
</details>
all but one was removed, as discussed, working on removing the final one.

---

*[storage/client_test.go, line 373 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPUNfBqy0hOrULK-S0k:-KPUUwDs6cTVUW6cwVOm:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/cdb6d126af4bc49f1bf6b5a020c76c92e9a7175b/storage/client_test.go#L373)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> "the those"
</details>
Done.

---

*[storage/client_test.go, line 383 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPUNvz0dLTB6nPHzvOY:-KPUV-XGxrvTCIEn5qJi:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/cdb6d126af4bc49f1bf6b5a020c76c92e9a7175b/storage/client_test.go#L383)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> replace "3" with `%d` `len(stores)`
</details>
Done.

---

*[storage/client_test.go, line 389 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPUO1JU97QnPIVkMENp:-KPUV0FGshtwFPn_DE-Z:bgmo1me) ([raw file](https://github.com/cockroachdb/cockroach/blob/cdb6d126af4bc49f1bf6b5a020c76c92e9a7175b/storage/client_test.go#L389)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> might remove this
</details>
I find this quite helpful actually when debugging.  Nice to know when the test has really started.

---

*[storage/store_pool.go, line 472 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPUHvMknac6AsYIxnq5:-KPUXpJDClKdhWyfbXSs:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/cdb6d126af4bc49f1bf6b5a020c76c92e9a7175b/storage/store_pool.go#L472)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> since this is only exported for tests, consider exporting it via helpers_test.go
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668)*

**Comment by tamird** [comment]
Reviewed 6 of 6 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[storage/client_raft_test.go, line 741 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPUNE8TBA8nxOJzw9pE:-KPUYxwKDDSq3MdJo6ZC:btojbfq) ([raw file](https://github.com/cockroachdb/cockroach/blob/5d09ed157cfbb6f4d8b7d736d335da6f229fc738/storage/client_raft_test.go#L741)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary>
> all but one was removed, as discussed, working on removing the final one.
</details>
Great.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668)*

**Comment by BramGruneir** [comment]
Review status: 2 of 4 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[storage/client_raft_test.go, line 741 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPUNE8TBA8nxOJzw9pE:-KPs4Ueixy7x1Gnl2qdk:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5d09ed157cfbb6f4d8b7d736d335da6f229fc738/storage/client_raft_test.go#L741)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> Great.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668)*

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

*[storage/client_raft_test.go, line 1620 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPs8R7JSHnpFMzGjV6k:-KPs8R7JSHnpFMzGjV6l:b-37819q) ([raw file](https://github.com/cockroachdb/cockroach/blob/a558ba323c23b0dc411f83c80df6c9012a75681d/storage/client_raft_test.go#L1620)):*
> ```Go
> 	}
> 
> 	// Replicate the 1st range to the stores 2 and 3.
> ```

this comment is explaining what is being done, which can be seen from the code, rather than why it is being done. The intention here is to raise the number of replicas of range 1 to 3, right?

---

*[storage/client_raft_test.go, line 1624 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPs8ctxcG6ILEBis_v6:-KPs8ctxcG6ILEBis_v7:b5ronsk) ([raw file](https://github.com/cockroachdb/cockroach/blob/a558ba323c23b0dc411f83c80df6c9012a75681d/storage/client_raft_test.go#L1624)):*
> ```Go
> 	mtc.replicateRange(replica1.Desc().RangeID, 1, 2)
> 
> 	// Replicate the 2nd range to stores 2, 4 and 5 and remove it from store 1.
> ```

ditto - explain the reason rather than repeating the code.

---

*[storage/client_raft_test.go, line 1625 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPs8BXeLY-VPHKoV3RO:-KPs8BXf8ZTDvGsMgF45:b-it8atz) ([raw file](https://github.com/cockroachdb/cockroach/blob/a558ba323c23b0dc411f83c80df6c9012a75681d/storage/client_raft_test.go#L1625)):*
> ```Go
> 
> 	// Replicate the 2nd range to stores 2, 4 and 5 and remove it from store 1.
> 	replica2 := mtc.stores[0].LookupReplica(roachpb.RKey("b"), nil)
> ```

extract local "b"

---

*[storage/client_raft_test.go, line 1647 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPs98sJjrcbTXRXMhb5:-KPs98sKZiwDmHDKsNg8:blrnam2) ([raw file](https://github.com/cockroachdb/cockroach/blob/a558ba323c23b0dc411f83c80df6c9012a75681d/storage/client_raft_test.go#L1647)):*
> ```Go
> 		t.Fatalf("expected store 1 to have %d replicas, actual %d", e, a)
> 	}
> 	if e, a := 2, startCounts[roachpb.StoreID(2)]; e != a {
> ```

this would be more robust as a comparison to a map literal - i think you can use pretty.Diff to render the diff.

---

*[storage/client_raft_test.go, line 1665 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPs9PGcAt0rmFKkIQ3f:-KPs9PGdPG151dTM9Csf:b-5uxyc) ([raw file](https://github.com/cockroachdb/cockroach/blob/a558ba323c23b0dc411f83c80df6c9012a75681d/storage/client_raft_test.go#L1665)):*
> ```Go
> 	mtc.initGossipNetwork()
> 
> 	util.SucceedsSoon(t, func() error {
> ```

can you add a comment explaining why this much code is inside the retry loop?

---

*[storage/client_test.go, line 1038 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPs9gjiMlUpJIQFmhcr:-KPs9gjiMlUpJIQFmhcs:bjeb31j) ([raw file](https://github.com/cockroachdb/cockroach/blob/a558ba323c23b0dc411f83c80df6c9012a75681d/storage/client_test.go#L1038)):*
> ```Go
> 	}
> 
> 	if err := origRepl.AdminTransferLease(destStore.Ident.StoreID); err != nil {
> ```

return origRepl.AdminTransfer....

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-:-KPs9xyfce3dlMtbiHHk:bnfp4nl)*

**Comment by tamird** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

*[storage/client_raft_test.go, line 1620 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPs8R7JSHnpFMzGjV6k:-KPsARK1XxMgBl_6uYtB:bsigj8c) ([raw file](https://github.com/cockroachdb/cockroach/blob/a558ba323c23b0dc411f83c80df6c9012a75681d/storage/client_raft_test.go#L1620)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> this comment is explaining what is being done, which can be seen from the code, rather than why it is being done. The intention here is to raise the number of replicas of range 1 to 3, right?
</details>
Actually, this block and the following block are pursuing the same result; a specific arrangement of replicas. Perhaps a small ascii diagram would best explain the intent.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668)*

**Comment by BramGruneir** [comment]
Review status: 2 of 4 files reviewed at latest revision, 6 unresolved discussions, some commit checks pending.

---

*[storage/client_raft_test.go, line 1620 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPs8R7JSHnpFMzGjV6k:-KPsM_yUnma4HxuXWUM1:bolvrjf) ([raw file](https://github.com/cockroachdb/cockroach/blob/a558ba323c23b0dc411f83c80df6c9012a75681d/storage/client_raft_test.go#L1620)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> Actually, this block and the following block are pursuing the same result; a specific arrangement of replicas. Perhaps a small ascii diagram would best explain the intent.
</details>
Yeah, I had one, but wasn't sure it was required and removed it.  I'll add it back in.

---

*[storage/client_raft_test.go, line 1624 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPs8ctxcG6ILEBis_v6:-KPsYq6R3lgXsliDWG6D:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a558ba323c23b0dc411f83c80df6c9012a75681d/storage/client_raft_test.go#L1624)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> ditto - explain the reason rather than repeating the code.
</details>
Done.

---

*[storage/client_raft_test.go, line 1625 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPs8BXeLY-VPHKoV3RO:-KPsZ8C874-M-ncqH3Ri:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a558ba323c23b0dc411f83c80df6c9012a75681d/storage/client_raft_test.go#L1625)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> extract local "b"
</details>
Done.

---

*[storage/client_raft_test.go, line 1647 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPs98sJjrcbTXRXMhb5:-KPsYpJrGaq1boGxQ8bJ:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a558ba323c23b0dc411f83c80df6c9012a75681d/storage/client_raft_test.go#L1647)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> this would be more robust as a comparison to a map literal - i think you can use pretty.Diff to render the diff.
</details>
Done.

---

*[storage/client_raft_test.go, line 1665 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPs9PGcAt0rmFKkIQ3f:-KPsZ96YwrVTONDFX8vt:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a558ba323c23b0dc411f83c80df6c9012a75681d/storage/client_raft_test.go#L1665)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> can you add a comment explaining why this much code is inside the retry loop?
</details>
Done.

---

*[storage/client_test.go, line 1038 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPs9gjiMlUpJIQFmhcr:-KPsZDY64Z4t7vJyJ4yu:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a558ba323c23b0dc411f83c80df6c9012a75681d/storage/client_test.go#L1038)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> return origRepl.AdminTransfer....
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668)*

**Comment by tamird** [comment]
Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[storage/client_raft_test.go, line 1628 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPseQ5X7dGZQcmeCm_H:-KPseQ5X7dGZQcmeCm_I:b-qxqdep) ([raw file](https://github.com/cockroachdb/cockroach/blob/7cbf2dc9d40804ee665f76e699776a7e2bcb3057/storage/client_raft_test.go#L1628)):*
> ```Go
> 	mtc.replicateRange(replica1.Desc().RangeID, 1, 2)
> 
> 	replica2Key := roachpb.RKey("b")
> ```

i should have been more specific - this is also the split key. please extract a local that is used in splitArgs as well.

---

*[storage/client_raft_test.go, line 1670 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPseioMuqBOJB4eM1jI:-KPseioMuqBOJB4eM1jJ:bc1phal) ([raw file](https://github.com/cockroachdb/cockroach/blob/7cbf2dc9d40804ee665f76e699776a7e2bcb3057/storage/client_raft_test.go#L1670)):*
> The leases tend to enjoy moving their way onto range 2

There's something wrong with this fragment.

---

*[storage/client_raft_test.go, line 1684 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPsfDJ2Js56xiKXzZNb:-KPsfDJ2Js56xiKXzZNc:b-vmd762) ([raw file](https://github.com/cockroachdb/cockroach/blob/7cbf2dc9d40804ee665f76e699776a7e2bcb3057/storage/client_raft_test.go#L1684)):*
> ```Go
> 
> 		// Gossip the stores so that the store pools are up to date. Note that
> 		// there may be a slight delay after gossiping has occurred.
> ```

this is unclear - delay between which two things?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668)*

**Comment by BramGruneir** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[storage/client_raft_test.go, line 1628 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPseQ5X7dGZQcmeCm_H:-KPsg8qhXMkg6On4XwEm:bf09sue) ([raw file](https://github.com/cockroachdb/cockroach/blob/7cbf2dc9d40804ee665f76e699776a7e2bcb3057/storage/client_raft_test.go#L1628)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> i should have been more specific - this is also the split key. please extract a local that is used in splitArgs as well.
</details>
splitArgs requires a Key, while the replica lookups require an rkey.

---

*[storage/client_raft_test.go, line 1670 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPseioMuqBOJB4eM1jI:-KPsgeTafTynqzkwKXYv:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/7cbf2dc9d40804ee665f76e699776a7e2bcb3057/storage/client_raft_test.go#L1670)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> > The leases tend to enjoy moving their way onto range 2
> 
> There's something wrong with this fragment.
</details>
Done.

---

*[storage/client_raft_test.go, line 1684 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPsfDJ2Js56xiKXzZNb:-KPsgwAWxJkGTKnBjVOh:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/7cbf2dc9d40804ee665f76e699776a7e2bcb3057/storage/client_raft_test.go#L1684)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> this is unclear - delay between which two things?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668)*

**Comment by BramGruneir** [comment]
Review status: 3 of 4 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[storage/client_raft_test.go, line 1628 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPseQ5X7dGZQcmeCm_H:-KPsmJqMFd4PM1XcoPUC:b-qc4jkr) ([raw file](https://github.com/cockroachdb/cockroach/blob/7cbf2dc9d40804ee665f76e699776a7e2bcb3057/storage/client_raft_test.go#L1628)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary>
> splitArgs requires a Key, while the replica lookups require an rkey.
</details>
I've combined them.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668)*

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[storage/client_raft_test.go, line 1670 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPseioMuqBOJB4eM1jI:-KPsmHvH3FImWCyqQasS:b-cgp61m) ([raw file](https://github.com/cockroachdb/cockroach/blob/7cbf2dc9d40804ee665f76e699776a7e2bcb3057/storage/client_raft_test.go#L1670)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary>
> Done.
</details>
Can you reword this to explain why the lease needs to be on store 1? Having done that, I don't think we need to explain why we're forcing the lease to be on that store.

---

*[storage/client_raft_test.go, line 1684 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPsfDJ2Js56xiKXzZNb:-KPsm8v2BQ7XY-qAsfPs:b-nnab07) ([raw file](https://github.com/cockroachdb/cockroach/blob/7cbf2dc9d40804ee665f76e699776a7e2bcb3057/storage/client_raft_test.go#L1684)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary>
> Done.
</details>
This sentence is broken. Please proof before resubmitting.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668)*

**Comment by BramGruneir** [comment]
Review status: 3 of 4 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[storage/client_raft_test.go, line 1670 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPseioMuqBOJB4eM1jI:-KPssCvCO8E16MOc75AL:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/7cbf2dc9d40804ee665f76e699776a7e2bcb3057/storage/client_raft_test.go#L1670)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> Can you reword this to explain why the lease needs to be on store 1? Having done that, I don't think we need to explain why we're forcing the lease to be on that store.
</details>
Done.

---

*[storage/client_raft_test.go, line 1684 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPsfDJ2Js56xiKXzZNb:-KPssDUtoyLj5VH51qR3:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/7cbf2dc9d40804ee665f76e699776a7e2bcb3057/storage/client_raft_test.go#L1684)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> This sentence is broken. Please proof before resubmitting.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668)*

**Comment by tamird** [comment]
:lgtm: again

---

Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[storage/client_raft_test.go, line 1670 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPseioMuqBOJB4eM1jI:-KPsu_qbU0_YUg6c6sRu:bg936q7) ([raw file](https://github.com/cockroachdb/cockroach/blob/7cbf2dc9d40804ee665f76e699776a7e2bcb3057/storage/client_raft_test.go#L1670)):*
How about:

> As of this writing, replicas which hold their range's lease cannot be removed; forcefully transfer the lease to another store to allow store 2's replica to be removed.

---

*[storage/client_raft_test.go, line 1684 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPsfDJ2Js56xiKXzZNb:-KPsv6lr00LNyQ_T9FFh:b6sv3cj) ([raw file](https://github.com/cockroachdb/cockroach/blob/7cbf2dc9d40804ee665f76e699776a7e2bcb3057/storage/client_raft_test.go#L1684)):*
> // Note that
> 		// there might be a delay post gossiping before the store pools are
> 		// updated.

How about:
> Note that there might be a delay between the call below and the asynchronous update of the store pools.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-:-KPsvGWSY98c3i8NH0CX:b-865gvj)*

**Comment by BramGruneir** [comment]
Review status: 3 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[storage/client_raft_test.go, line 1670 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPseioMuqBOJB4eM1jI:-KPt1v2U15kPbiSSNuaO:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/7cbf2dc9d40804ee665f76e699776a7e2bcb3057/storage/client_raft_test.go#L1670)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> How about:
> 
> > As of this writing, replicas which hold their range's lease cannot be removed; forcefully transfer the lease to another store to allow store 2's replica to be removed.
</details>
Done.

---

*[storage/client_raft_test.go, line 1684 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668#-KPsfDJ2Js56xiKXzZNb:-KPt1cN1YUNDDJLf4tLj:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/7cbf2dc9d40804ee665f76e699776a7e2bcb3057/storage/client_raft_test.go#L1684)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> > // Note that
> > 		// there might be a delay post gossiping before the store pools are
> > 		// updated.
> 
> How about:
> > Note that there might be a delay between the call below and the asynchronous update of the store pools.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668)*

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8668)*
